src/Analysis/Ast/Impl/Analyzer/AnalysisWalker.cs

```diff
@@ -67,7 +69,7 @@ internal abstract class AnalysisWalker : PythonWalker {
                     AssignmentHandler.HandleAnnotatedExpression(ea, null);
                     return false;
                 case Comprehension comp:
-                    Eval.ProcessComprehension(comp);
+                    Eval.ProcessComprehensionInScope(comp, Eval.Interpreter.LanguageVersion.Is3x());
```

Comment:
Eval contains Interpreter. Why pass explicitly when it can look up the version itself?

Reply:
Could probably get away without changing the function name that way (smaller diff).

Reply:
I did that because basically for 3X, we have to open a scope then process the comprehension, and for 2X we just have to process the comprehension. I didn't want to have the duplicated code so I wrapped it in another method `ProcessComprehension` and had `ProcessComprehensionInScope` handle whether to open a new scope or not. Do you know any constructs that would be nicer?

Reply:
Keeping the same name and reading internally is fine though